### PR TITLE
[contract] change realm positions to make them closer to each other even if they are far away

### LINKED ISSUE #2093: [contract] change realm positions to make them closer to each other even if they are far away

### PR DESCRIPTION
**Pull request by aymericdelab**
Fixes #2093

## Summary by CodeRabbit

- **New Features**
	- Introduced a new public function `PI()` to return the mathematical constant π.
	- Restructured the `SettlementConfig` parameters for improved settlement management.

- **Bug Fixes**
	- Enhanced error handling for troop purchases in combat tests.

- **Documentation**
	- Updated comments and documentation to reflect changes in settlement configuration logic.

- **Tests**
	- Modified tests for army creation, battle mechanics, and settlement configurations to align with new parameters and logic.
	- Added new test function `config_test_settle_on_coords_with_structure` to validate settlement placement behavior.

### CODE REVIEW

File: contracts/src/systems/combat/tests/battle_leave_test.cairo [hunk 1]
```diff
@@ -94,14 +94,13 @@ fn set_configurations(ref world: WorldStorage) {
...
-                max_distance: 5,
-                min_scaling_factor_scaled: 1844674407370955161,
-                min_angle_increase: 30,
-                max_angle_increase: 100,
+                base_distance: 10,
+                min_first_layer_distance: 30,
+                points_placed: 0,
+                current_layer: 1,
+                current_side: 1,
+                current_point_on_side: 0,
```
Comment (edisontim) [inline contracts/src/systems/combat/tests/battle_leave_test.cairo:103]: Could you write a default config and put that in the testing default config? It'll be clearer to have a function that returns it :)

**Review by edisontim — CHANGES_REQUESTED** [review]

**Review by ponderingdemocritus — APPROVED** [review]
lgtm

File: sdk/packages/eternum/src/constants/global.ts [hunk 2]
```diff
@@ -226,14 +225,13 @@ export const EternumGlobalConfig: Config = {
...
-    max_distance: SETTLEMENT_MAX_DISTANCE,
-    min_scaling_factor_scaled: SETTLEMENT_MIN_SCALING_FACTOR_SCALED,
-    min_angle_increase: SETTLEMENT_MIN_ANGLE_INCREASE,
-    max_angle_increase: SETTLEMENT_MAX_ANGLE_INCREASE,
+    base_distance: SETTLEMENT_BASE_DISTANCE,
+    min_first_layer_distance: SETTLEMENT_MIN_FIRST_LAYER_DISTANCE,
+    points_placed: SETTLEMENT_POINTS_PLACED,
+    current_layer: SETTLEMENT_CURRENT_LAYER,
+    current_side: SETTLEMENT_CURRENT_SIDE,
+    current_point_on_side: SETTLEMENT_CURRENT_POINT_ON_SIDE,
```
[the remark this answers is not in the document]
Comment (ponderingdemocritus) [reply sdk/packages/eternum/src/constants/global.ts:234]: @aymericdelab implement this

[the remark this answers is not in the document]
Comment (ponderingdemocritus) [reply sdk/packages/eternum/src/constants/global.ts:234]: good bot

**Pull request merged**